dataprofiler/profilers/numerical_column_stats.py

```diff
@@ -321,11 +346,14 @@ def _get_histogram(self, values, bin_method):
                 unique_value = values.iloc[0]
             bin_edges = np.array([unique_value, unique_value])
         else:
-            values, weights = histogram_utils._ravel_and_check_weights(
-                values, None)
-            _, n_equal_bins = histogram_utils._get_bin_edges(
-                values, bin_method, None, None)
-            n_equal_bins = min(n_equal_bins, self.max_histogram_bin)
+            n_equal_bins = suggested_bin_count = self.user_set_histogram_bin
```

Comment:
There is no reason to define suggested_bin_count here since it is only used within the if statement after a second definition. Also I think the naming between n_equal_bins and suggested_bin_count isn't super clear considering suggested_bin_count seems to just be an intermediary value to ensure n_equal_bins lands between the min and max histogram bins.

Reply:
This is actually an intermediary step before the next PR which technically exists: https://github.com/capitalone/DataProfiler/pull/144

```python
else:
    n_equal_bins = suggested_bin_count = self.user_set_histogram_bin
    if n_equal_bins is None:
        values, weights = histogram_utils._ravel_and_check_weights(
            values, None)
        _, n_equal_bins = histogram_utils._get_bin_edges(
            values, bin_method, None, None)
        suggested_bin_count = min(n_equal_bins, self.max_histogram_bin)
        n_equal_bins = max(self.min_histogram_bin, suggested_bin_count)
    bin_counts, bin_edges = np.histogram(values, bins=n_equal_bins)
return bin_counts, bin_edges, suggested_bin_count
```
the reason being is if we don't have a user set number of bins. we use the min/max bin count to choose the bin count and the suggested bin count by each method (which has to be less than the max) is passed up to be stored. 

That way at report time, we return the best histogram rather than a really sparse one that may have more bins than necessary. e.g. min bins = 1000, but only 5 numbers. we probably only need 3-5 bins at report otherwise the histogram would be no bueno. So we need to know the suggested to convert from the 1000 -> 3-5 bins

Reply:
I'm open to naming changes, but maybe we change in next PR since we will be adapting this code anyways and it is not functionally harming anything